lib/server.ml

```diff
@@ -139,6 +139,15 @@ let connect channel ?offer () =
         else negotiate_tls make_tls_channel
       )
 
+let with_connection clearchan ?offer f =
+  (match offer with
+    | None -> connect clearchan ()
+    | Some offer -> connect clearchan ~offer ())
```

Comment:
Luckily we can simplify propagating optional parameters with:
`connect clearchan ?offer ()` :smile:

Reply:
Oops, yes, that's why the function has that last parameter.